### PR TITLE
Donut: Optimize donut contrast for dark/light mode

### PR DESCRIPTION
**Pull request by flourish86**
refs #100

rainbow-dark
rainbow-light

### CODE REVIEW

**Comment by yhabteab** [comment]
Bildschirm­foto 2023-02-09 um 10 31 29

**Comment by nilmerg** [comment]
Seems fine, by just looking at the shots above. Though, I wonder why the donuts aren't styled with CSS in the first place? Wouldn't this avoid future issues with modes/themes also?

File: library/X509/Donut.php [hunk 1]
```diff
@@ -73,16 +73,14 @@ public function setLabelCallback(callable $callback)
...
 
-        foreach ($this->data as $data) {
-            $color = $colorScheme();
-            $donut->addSlice((int) $data['cnt'], ['stroke' => $color]);
+        foreach ($this->data as $index => $data) {
+            $donut->addSlice((int) $data['cnt'], ['class' => 'segment-' . $index]);
             $legend->addRow(
                 [
-                    Html::tag('span', ['class' => 'badge', 'style' => "background-color: $color; height: 1.75em;"]),
+                    Html::tag('span', ['class' => 'badge badge-' . $index, 'style' => "height: 1.75em;"]),
```
Comment (nilmerg) [inline library/X509/Donut.php:83]: Please also transfer the `height` definition. Using `style` is discouraged.

File: public/css/module.less [hunk 2]
```diff
@@ -70,8 +70,58 @@
 }
 
 .cert-donut {
+  @segment-color-0: #FF595E;
+  @segment-color-1: #FFCA3A;
+  @segment-color-2: #8AC926;
+  @segment-color-3: #1982C4;
+  @segment-color-4: #6A4C93;
```
Comment (nilmerg) [inline public/css/module.less:77]: This way it's scoped to `.cert-donut`. Please put them into the module's global scope.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: public/css/module.less (hunk 2 above)
Comment (flourish86) [reply public/css/module.less]: ✅

File: library/X509/Donut.php (hunk 1 above)
Comment (flourish86) [reply library/X509/Donut.php]: ✅

File: public/css/module.less [hunk 3]
```diff
@@ -1,5 +1,11 @@
 // Icinga Web 2 X.509 Module | (c) 2018 Icinga GmbH | GPLv2
 
+@segment-color-0: #FF595E;
+@segment-color-1: #FFCA3A;
+@segment-color-2: #8AC926;
+@segment-color-3: #1982C4;
+@segment-color-4: #6A4C93;
```
Comment (nilmerg) [inline public/css/module.less:7]: I guess your intention by putting this into `.cert-donut`, was that it emphasizes their meaning? Nothing speaks against a more specific prefix then, e.g. `@cert-donut-segment-0`.

**Comment by yhabteab** [comment]
```php
variable @segment-color-0 is undefined in file anonymous-file-0.less in anonymous-file-0.less
#0 /icingaweb2/library/Icinga/Less/ColorPropOrVariable.php(56): Less_Tree_Variable->compile(Object(Less_Environment))
#1 /icingaweb2/library/vendor/lessphp/lib/Less/Tree/Expression.php(51): Icinga\Less\ColorPropOrVariable->compile(Object(Less_Environment))
#2 /icingaweb2/library/vendor/lessphp/lib/Less/Tree/Value.php(26): Less_Tree_Expression->compile(Object(Less_Environment))
#3 /icingaweb2/library/vendor/lessphp/lib/Less/Tree/Rule.php(73): Less_Tree_Value->compile(Object(Less_Environment))
#4 /icingaweb2/library/vendor/lessphp/lib/Less/Tree/Ruleset.php(91): Less_Tree_Rule->compile(Object(Less_Environment))
```

**Comment by yhabteab** [comment]
Looks like you renamed the variables but forgot to adjust all the usages.

**Comment by nilmerg** [comment]
:scream:

**Comment by lippserd** [comment]
Boo.

**Pull request merged**